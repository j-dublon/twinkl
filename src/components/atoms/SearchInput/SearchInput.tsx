import { FC } from "react";
import { SearchInputProps } from "@/types";

export const SearchInput: FC<SearchInputProps> = ({ value, onChange }) => {
  return (
    <div className="bg-zinc-100 py-2 sm:bg-white focus:outline-none focus:ring-2 focus:ring-blue-500">
      <input
        placeholder="Search..."
        onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
          onChange(e.target.value)
        }
        value={value}
        type="search"
        className="block m-auto bg-white sm:bg-zinc-100 p-2 rounded-sm w-90/100 sm:w-40/100 lg:w-30/100"
      />
    </div>
  );
};
