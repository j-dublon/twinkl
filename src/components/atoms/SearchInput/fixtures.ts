import { SearchInputProps } from "@/types";
import { Mock } from "vitest";

export const mockOnChange: Mock = vi.fn();

export const mockProps: SearchInputProps = {
  value: "",
  onChange: mockOnChange,
};
