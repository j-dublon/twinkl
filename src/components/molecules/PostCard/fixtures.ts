import { Post } from "@/types";

export const mockPost: Post = {
  id: 1,
  title: "A great post",
  body: "Some information about this post",
  userId: 1,
};
