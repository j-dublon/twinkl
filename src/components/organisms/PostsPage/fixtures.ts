import { Mock } from "vitest";

export const mockDeletePost: Mock = vi.fn();
