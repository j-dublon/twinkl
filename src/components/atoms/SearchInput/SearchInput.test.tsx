import { fireEvent, render } from "@testing-library/react";
import { SearchInput } from "./SearchInput";
import { mockOnChange, mockProps } from "./fixtures";

describe("Component: SearchInput", () => {
  it("SHOULD match snapshot", () => {
    const component = render(<SearchInput {...mockProps} />);
    expect(component).toMatchSnapshot();
  });

  it("SHOULD call onChange function WHEN user types into the input", () => {
    const mockQuery = "Something";
    const { getByPlaceholderText } = render(<SearchInput {...mockProps} />);

    fireEvent.change(getByPlaceholderText("Search..."), {
      target: { value: mockQuery },
    });
    expect(mockOnChange).toHaveBeenCalledWith(mockQuery);
  });
});
