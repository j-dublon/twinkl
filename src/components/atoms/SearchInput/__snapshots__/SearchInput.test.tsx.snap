// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: SearchInput > SHOULD match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-zinc-100 py-2 sm:bg-white focus:outline-none focus:ring-2 focus:ring-blue-500"
      >
        <input
          class="block m-auto bg-white sm:bg-zinc-100 p-2 rounded-sm w-90/100 sm:w-40/100 lg:w-30/100"
          placeholder="Search..."
          type="search"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-zinc-100 py-2 sm:bg-white focus:outline-none focus:ring-2 focus:ring-blue-500"
    >
      <input
        class="block m-auto bg-white sm:bg-zinc-100 p-2 rounded-sm w-90/100 sm:w-40/100 lg:w-30/100"
        placeholder="Search..."
        type="search"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
